vnext/Microsoft.ReactNative/Views/ReactRootControl.cpp

```diff
@@ -607,6 +608,53 @@ void ReactRootControl::ReloadViewHost() noexcept {
   }
 }
 
+void ReactRootControl::AttachBackHandlers(XamlView const &rootView) noexcept {
+  auto rootElement(rootView.as<winrt::UIElement>());
+  if (rootElement == nullptr) {
+    assert(false);
+    return;
+  }
+
+  winrt::Windows::UI::Core::SystemNavigationManager::GetForCurrentView().BackRequested(
+      [this](winrt::IInspectable const & /*sender*/, winrt::BackRequestedEventArgs const &args) {
+        args.Handled(OnBackRequested());
+      });
+
+  // Handle keyboard "back" button press
+  winrt::KeyboardAccelerator goBack{};
```

Comment:
I'm not sure this is the right way to listen for the back action. From what I can find, `SystemNavigationManager` has a `BackRequested` event to express this intent. Can we hook into that instead?

Reply:
The `BackRequested` event does not fire in all circumstances. The [linked documentation](https://docs.microsoft.com/en-us/windows/uwp/design/basics/navigation-history-and-backwards-navigation?redirectedfrom=MSDN#optimizing-for-different-device-and-form-factors) in the description outlines this in a table:
![image](https://user-images.githubusercontent.com/687190/79508025-a0349a00-7fed-11ea-9bb4-8c3bb6e6ea60.png)

Reply:
Oh wow, you're right. I completely missed that in the description and in code. Thanks for linking it again.

It's unfortunate official reccomendations stray from the system API, but this makes sense then. I wonder a bit whether we could let developers opt into non system back events, but could see arguments both ways.

Reply:
Consider to add a comment with a link.

---
In reply to: [409855996](https://github.com/microsoft/react-native-windows/pull/4623#discussion_r409855996) [](ancestors = 409855996)

Reply:
Added a comment with a link to the docs